c10/core/DispatchKey.h

```diff
@@ -76,6 +76,13 @@ enum class DispatchKey : uint8_t {
   OpenCL,
   IDEEP,
 
+  // A meta tensor is a tensor without any data associated with it.  (They
+  // have also colloquially been referred to as tensors on the "null" device).
+  // A meta tensor can be used to dry run operators without actually doing any
+  // computation, e.g., add on two meta tensors would give you another meta
+  // tensor with the output shape and dtype, but wouldn't actually add anything.
+  Meta,
```

Comment:
It seems like the old comment here tried to explain the differences between a meta tensor and a meta function (which is called by both normal and meta tensors), which I thought was pretty helpful context.

Reply:
The comment was hopelessly out of date though :) I can bring back some new text explaining the difference between a meta function and a meta tensor, though